FirstLast.java

```diff
@@ -0,0 +1,54 @@
+// Time Complexity : O(log n) since we are doing binary search recursively
+// Space Complexity : O(log n) number of recursive calls
+// Did this code successfully run on Leetcode : Partially
+// Any problem you faced while coding this : Faling for some cases. There are different possibilities once I search the required element.
+/* The previous element can be equal or the next can be equal or both the prev and next can be equal to search element. Challenge was to 
+consider all the possibilities. 
+Algorithm:  Find the mid element recursively to land up on the searched element. Then do a binary search recursively at the left and binary search
```

Comment:
We need to eliminate half elements always. If you find yourself doing binary search on both side then it is as efficient as linear search